app/src/main/java/org/fossasia/openevent/general/event/EventUtils.kt

```diff
@@ -51,12 +53,27 @@ object EventUtils {
     fun loadMapUrl(event: Event) = "geo:<${event.latitude}>,<${event.longitude}>" +
         "?q=<${event.latitude}>,<${event.longitude}>"
 
-    fun getLocalizedDateTime(dateString: String): ZonedDateTime = ZonedDateTime.parse(dateString)
-            .toOffsetDateTime()
-            .atZoneSameInstant(ZoneId.systemDefault())
+    fun getEventDateTime(dateString: String, timeZone: String): ZonedDateTime {
+        try {
+            return when (PreferenceManager.getDefaultSharedPreferences(OpenEventGeneral.appContext)
+                .getBoolean("useEventTimeZone", false)) {
+
+                true -> ZonedDateTime.parse(dateString)
+                    .toOffsetDateTime()
+                    .atZoneSameInstant(ZoneId.of(timeZone))
+                false -> ZonedDateTime.parse(dateString)
+                    .toOffsetDateTime()
+                    .atZoneSameInstant(ZoneId.systemDefault())
+            }
+        } catch (e: NullPointerException) {
```

Comment:
Where can null pointer exception happen?

Reply:
I get that error when I run testFdroidDebugUnitTest

Reply:
This must be returning null PreferenceManager.getDefaultSharedPreferences(OpenEventGeneral.appContext).getBoolean("useEventTimeZone", false)

Reply:
Then fix that

Reply:
This is because OpenEventGeneral.appContext is null in testFdroidDebugUnitTest. It's not null when app is running normally. I don't think this can be fixed. I checked.

Reply:
> This is because OpenEventGeneral.appContext is null when running testFdroidDebugUnitTest. It's not null when app is running normally. I don't think this can be fixed. I checked.

Reply:
Do I need to change anything here?